source/Server.Extensibility/Extensions/ExtResult.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Octopus.Server.Extensibility.Extensions
+{
+    public interface IExtResult
```

Comment:
I'm not too sure on this name...maybe `IValueResult` instead?

_I know this means we'd have to make the interface generic and add the `Value` property to the interface (instead of `ExtResult<T>` below) and removing the `FailureResult` struct, but is that a bad thing? I think it'd make it clearer what we're getting when calling the method._

_**Or**_, even though we have the `IResult/Result` class in `Octopus.Core` (and it's working on the same principle as this implementation), couldn't we name this object `IResult` as well, or you think it'll be confusing when trying to figure out which on to use? 
In the extensions world we'll only ever know of this `Result` type since we can't see the one in `Octopus.Core`, and if you're in Octopus would you think to use the one that's in `ServerExtensibility` if you're working on code that isn't to do with extensibility...?

Reply:
Renamed.